File: Python/chapitre2/Exercice5.py
import os
import random
import time

debut = time.time()
p1 = os.fork()

if p1 == 0:
    time.sleep(random.randint(1, 10))
    os._exit(0)
p2 = os.fork()
if p2 == 0:
    time.sleep(random.randint(1, 10))
    os._exit(0)
os.waitpid(p1, 0)
os.waitpid(p2, 0)
fin = time.time()

print("La durée total : {} secondes ".format(int(fin - debut)))

File: C/chapitre2/Exercice5.c
#include <stdio.h>
#include <unistd.h>
#include <sys/wait.h>
#include <time.h>
#include <stdlib.h>

int main(void)
{
    time_t debut;
    time_t fin;
    time(&debut);

    srand(time(NULL));
    pid_t p1 = fork();
    pid_t p2 = fork();

    if(p1 == 0){
        sleep(rand() % 10 + 1);
        _exit(0);
    }
    if (p2 == 0){
        sleep(rand() % 10 + 1);
        _exit(0);
    }
    wait(NULL);
    wait(NULL);
    time(&fin);
    printf("La durée total %ld secondes", fin - debut);
    return (0);
}